Process_Library/GOCART2G_MieMod.F90

```diff
@@ -558,18 +561,22 @@ real function getWavelength(this, ith_channel, rc) result (wavelength)
      real, parameter :: w_tol = 1.e-9
      integer :: i
 
-     if (present(rc)) rc = 0
-
      if (ith_channel <=0 .or. ith_channel > this%nch ) then
-       !$omp critical (GetWav)
-       print*, "The channel of ",ith_channel, " is an invalid channel number."
-       !$omp end critical (GetWav)
-       if (present(rc)) rc = -1
-       wavelength = -1. ! meanlingless nagative
-       return
+        wavelength = -1. ! meaningless negative
+     else
+        wavelength = this%wavelengths(ith_channel)
      endif
-     
-     wavelength = this%wavelengths(ith_channel)
+
+    if (present(rc)) then
+       if (wavelength > 0) then
+          rc = __SUCCESS__
+       else
+          rc = __FAIL__
+          !$omp critical (GetWav)
+          print*, "The channel of ",ith_channel, " is an invalid channel number."
+          !$omp end critical (GetWav)
+       endif
+    endif
```

Comment:
I believe that the following is equivalent and cleaner.
```suggestion
    _FAIL('Illegal value for ith_channel.')
```

Reply:
Never mind.  Regular MAPL macros not available here.

Reply:
We could add the missing macros to Process.H and enable them if the process library is built without MAPL.